src/lib/signature_kind/compile_config/mina_signature_kind.ml

```diff
@@ -2,11 +2,11 @@
 
 type t = Testnet | Mainnet
 
-[%%if mainnet]
+[%%if network = "mainnet"]
 
 let t = Mainnet
 
-[%%else]
+[%%elif network = "testnet"]
```

Comment:
Would it be useful to have an `else` case that generates a compile-time error, if `network` is neither "mainnet" nor "testnet"? That would catch a bug in the config file.

Reply:
Matthew was asking me to add another variant `Other_of_string` to catch all the other cases and I pushed a commit to do that.